clang/test/Driver/sycl-offload.c

```diff
@@ -171,6 +171,19 @@
 
 /// ###########################################################################
 
+/// Check -Xsycl-target-frontend= accepts triple aliases
+// RUN:   %clang -### -fsycl -fsycl-targets=spir64 -Xsycl-target-frontend=spir64 -DFOO %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-UNUSED-ARG-WARNING %s
+// RUN:   %clang -### -fsycl -fsycl-targets=spir64_x86_64 -Xsycl-target-frontend=spir64_x86_64 -DFOO %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-UNUSED-ARG-WARNING %s
+// RUN:   %clang -### -fsycl -fsycl-targets=spir64_gen -Xsycl-target-frontend=spir64_gen -DFOO %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-UNUSED-ARG-WARNING %s
+// RUN:   %clang -### -fsycl -fsycl-targets=spir64_fpga -Xsycl-target-frontend=spir64_fpga -DFOO %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-UNUSED-ARG-WARNING %s
+// CHK-UNUSED-ARG-WARNING-NOT: clang{{.*}} warning: argument unused during compilation: '-Xsycl-target-frontend={{.*}} -DFOO'
+
```

Comment:
I'd suggest adding a positive test case as well, at least ensuring that the propagation of the target option still happens. Something along the lines of:
```suggestion
// RUN:   %clang -### -fsycl -fsycl-targets=spir64 -Xsycl-target-frontend=spir64 -DFOO %s 2>&1 \
// RUN:   | FileCheck -check-prefix=CHK-TARGET-FRONTEND-ALIAS %s
// RUN:   %clang -### -fsycl -fsycl-targets=spir64_x86_64 -Xsycl-target-frontend=spir64_x86_64 -DFOO %s 2>&1 \
// RUN:   | FileCheck -check-prefix=CHK-TARGET-FRONTEND-ALIAS %s
// RUN:   %clang -### -fsycl -fsycl-targets=spir64_gen -Xsycl-target-frontend=spir64_gen -DFOO %s 2>&1 \
// RUN:   | FileCheck -check-prefix=CHK-TARGET-FRONTEND-ALIAS %s
// RUN:   %clang -### -fsycl -fsycl-targets=spir64_fpga -Xsycl-target-frontend=spir64_fpga -DFOO %s 2>&1 \
// RUN:   | FileCheck -check-prefix=TARGET-FRONTEND-ALIAS %s
// CHK-TARGET-FRONTEND-ALIAS-NOT: clang{{.*}} warning: argument unused during compilation: '-Xsycl-target-frontend={{.*}} -DFOO'
// CHK-TARGET-FRONTEND-ALIAS: clang{{.*}} "-cc1"{{.*}} "-fsycl-is-device"{{.*}} "-DFOO"{{.*}}

```

Reply:
Ok, I'll add a CHECK to check targets.